Warfare/Warfare/GameEngine.swift

```diff
@@ -10,13 +10,413 @@ class GameEngine {
     var game: Game!
     var map: Map { return self.game.map }
     var scene: GameScene?
-    
+
     init() { }
-    
+
     func newGame() {
         self.loadMap(number: "1")
     }
 
+    // MARK: - Operations
+
+    // Should get called once a cycle (after all player have played their turn
+    func growTrees() {
+        let allTiles = self.map.tiles.rows.reduce([], combine: +)
+        var seen = [Tile]()
+
+        for t in allTiles {
+            // Only visit unseen tiles
+            if contains(seen, { $0 === t}) { continue }
+            seen.append(t)
+
+            // Only consider tiles with trees
+            if t.land != .Tree { continue }
+
+            for n in self.map.neighbors(tile: t) {
+                if contains(seen, { $0 === n }) { continue }
+                seen.append(n)
+
+                if n.isGrowable() {
+                    let random = Int(arc4random_uniform(2))
```

Comment:
I'm not sure how "random" that neighbour really is provided that map.neighbors() returns the neighbours from top left to down left clockwise. But I get that you only check unseen tiles.